src/test/java/com/fasterxml/jackson/databind/deser/TestCustomSubTypeValidator.java

```diff
@@ -0,0 +1,277 @@
+            _name = name;
+        }
+
+        public List<Animal> getAnimals()
+        {
+            return _animals;
+        }
+
+        public void setAnimals(List<Animal> animals)
+        {
+            _animals = animals;
+        }
+    }
+
+    @JsonTypeInfo(
+        use=JsonTypeInfo.Id.CLASS,
+        include=JsonTypeInfo.As.PROPERTY,
+        property="species"
+    )
+    public static abstract class Animal
+    {
+        private String _name;
+
+        public String getName() {
+            return _name;
+        }
+
+        public void setName(String name) {
+            _name = name;
+        }
+    }
+
+    public static class Lion extends Animal
+    {
+    }
+
+    public static class Warthog extends Animal
+    {
+    }
+
+    public static class Goat extends Animal
+    {
+    }
+
+    public static class Sheep extends Animal
+    {
+    }
+
+    public static class Penguin extends Animal
+    {
+    }
+
+    /*
+    /**********************************************************
+    /* Helper classes (validators)
+    /**********************************************************
+     */
+
+    static class DenyAllValidator
+        extends SubTypeValidator
+    {
+        @Override
+        public void validateSubType(DeserializationContext ctxt, JavaType type, BeanDescription beanDesc)
+            throws JsonMappingException 
+        {
+            ctxt.reportBadTypeDefinition(
+                    beanDesc,
+                    "Illegal type (%s) to deserialize: this validator does not allow deserialization at all!",
+                    type.getRawClass().getName());
+        }
+    }
+
+    static class FriendlyAnimalValidator
+        extends SubTypeValidator
+    {
+        private static Class<?>[] FRIENDLY_ANIMAL_WHITELIST = new Class<?>[] {
+            Animal.class,
+            Goat.class,
+            Sheep.class,
+            Penguin.class
+        };
+
+        @Override
+        public void validateSubType(DeserializationContext ctxt, JavaType type, BeanDescription beanDesc)
+            throws JsonMappingException 
+        {
+            super.validateSubType(ctxt, type, beanDesc);
+
+            if (type.isTypeOrSubTypeOf(Animal.class)) {
+                for (final Class<?> friendlyAnimalSpecies : FRIENDLY_ANIMAL_WHITELIST) {
+                    if (type.hasRawClass(friendlyAnimalSpecies)) {
+                        return;
+                    }
+                }
+
+                ctxt.reportBadTypeDefinition(
+                        beanDesc,
+                        "Illegal type (%s) to deserialize: prevented because this animal species is not known to be friendly",
+                        type.getRawClass().getName());
+            }
+        }
+    }
+
+    /*
+    /**********************************************************
+    /* Factory methods
+    /**********************************************************
+     */
+
+    protected ObjectMapper objectMapper(SubTypeValidator validator)
```

Comment:
Fyi, this factory method demonstrates how an `ObjectMapper` could be configured with a custom `SubTypeValidator` in real-life.

I chose to use sub-class `JsonMapper`, because `ObjectMapper` does not seem to be configurable enough. Could that limit the use of custom `SubTypeValidators`? Is there a way to use `ObjectMapper` directly, after all?